dom.go

```diff
@@ -991,19 +1000,40 @@ func renderComponent(next Component, prev ComponentOrHTML) (h *HTML, skip bool,
 
 	// Render the component into HTML, handling nil renders.
 	nextRender := next.Render()
+	prevRender := next.Context().prevRender
 	if nextRender == nil {
 		// nil renders are translated into noscript tags.
 		nextRender = Tag("noscript")
 	}
 
-	// Reconcile the actual rendered HTML.
-	pendingMounts = nextRender.reconcile(extractHTML(prev))
+	switch v := nextRender.(type) {
+	case Component:
+		nextHTML, skip, pendingMounts = renderComponent(v, prevRender)
+		if skip {
+			return nextHTML, skip, pendingMounts
+		}
+	case *HTML:
+		if v == nil {
+			// nil renders are translated into noscript tags.
+			v = Tag("noscript")
+		}
+		nextHTML = v
+		// Reconcile the actual rendered HTML.
+		pendingMounts = nextHTML.reconcile(extractHTML(prev))
+	default:
+		panic("vecty: encountered invalid ComponentOrHTML " + reflect.TypeOf(v).String())
+	}
+
+	m := mountUnmount(nextRender, prevRender)
+	if m != nil {
+		pendingMounts = append(pendingMounts, m)
+	}
 
 	// Update the context to consider this render.
 	next.Context().prevRender = nextRender
-	next.Context().prevRenderComponent = doCopy(next)
+	next.Context().prevRenderComponent = copyComponent(next)
 	next.Context().unmounted = false
```

Comment:
Should we be setting `next.Context().mounted = true` here as well?

Reply:
It's not mounted until the actual mount is performed, but it can't be unmounted if we're rendering it.

Reply:
Hmm, got it. I'll have to look into this some more to understand. For now I trust you on this & will merge :)